pkg/utils/jsonutils.go

```diff
@@ -58,3 +60,29 @@ func FlattenSingleValue(key string, m map[string]interface{}, child interface{})
 		m[key] = child
 	}
 }
+
+// GetString extracts a string value from JSON data based on the provided keys.
+// It uses the supplied workBuf to avoid unnecessary allocations and resizing.
+// If the value is not a string or if an error occurs, it returns an appropriate error.
+func GetString(data []byte, workBuf []byte, keys ...string) ([]byte, error) {
+    v, dataType, _, err := jsonparser.Get(data, keys...)
+    if err != nil {
+        return nil, err
+    }
+
+    if dataType != jsonparser.String {
+        if dataType == jsonparser.Null {
+            return nil, fmt.Errorf("key %s has a null value", strings.Join(keys, ", "))
+        }
+        return nil, fmt.Errorf("value is not a string: %s", string(v))
+    }
+
+    if cap(workBuf) < len(v) {
+        workBuf = make([]byte, len(v))
```

Comment:
we should not erase out the old workbuf, but rather __extend__ it. Checkout examples of we have been using the `ResizeSlice` function in such cases

Reply:
updated